### PR TITLE
BUG: DataFrame reductions inconsistent with Series counterparts

### LINKED ISSUE #21020: Series[categorical] median raises, but DataFrame doesn't
This is a bit strange.

```python
In [3]: pd.DataFrame({"A": pd.Categorical([1, 2, 2, 2, 3])})
Out[3]:
   A
0  1
1  2
2  2
3  2
4  3

In [4]: df = pd.DataFrame({"A": pd.Categorical([1, 2, 2, 2, 3])})

In [5]: df.median()
Out[5]:
A    2.0
dtype: float64

In [6]: df.A.median()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-15196eaacc89> in <module>()
----> 1 df.A.median()

~/sandbox/pandas/pandas/core/generic.py in stat_func(self, axis, skipna, level, numeric_only, **kwargs)
   9587                                       skipna=skipna)
   9588         return self._reduce(f, name, axis=axis, skipna=skipna,
-> 9589                             numeric_only=numeric_only)
   9590
   9591     return set_function_name(stat_func, name, cls)

~/sandbox/pandas/pandas/core/series.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   3220         return delegate._reduce(op=op, name=name, axis=axis, skipna=skipna,
   3221                                 numeric_only=numeric_only,
-> 3222                                 filter_type=filter_type, **kwds)
   3223
   3224     def _reindex_indexer(self, new_index, indexer, copy):

~/sandbox/pandas/pandas/core/arrays/categorical.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   2065         if func is None:
   2066             msg = 'Categorical cannot perform the operation {op}'
-> 2067             raise TypeError(msg.format(op=name))
   2068         return func(numeric_only=numeric_only, **kwds)
   2069

TypeError: Categorical cannot perform the operation median
```

Anyone know whether that's intentional?

### LINKED ISSUE #28949: BUG: np.min on DataFrame with a non-ordered Categorical col
Discovered while debugging the try/excepts in groupby:

```
cat = pd.Categorical(['a', 'b', 'c', 'b'], ordered=False)
ser = pd.Series(cat)
df = ser.to_frame()
>>> np.min(cat)   # <-- correctly raises
>>> np.min(ser)   # <-- correctly raises
>>> np.min(df)     # <-- incorrectly returns Series(['a'])
```

### LINKED ISSUE #36076: BUG: ser.to_frame().all() inconsistent with ser.all() with CategoryDtype
Based on tests.frame.test_analytics.test_any_all_np_func

```
ser = pd.Series([0, 1], dtype="category", name="A")
df = ser.to_frame()

>>> ser.all().
TypeError: Categorical cannot perform the operation all

>>> df.all().item()
False
```

Whats happening here is the DataFrame case is calling `self.values` which casts to int64 (avoidable with 2D EAs...).

If instead we refactor (xref #35881) to operate blockwise, this block is ignored, so we end up getting the result we'd get on an empty DataFrame, i.e. True.  (xref #28900 ignoring failures is a footgun).

Expected Behavior: as long as have ignore_failures, `df.all()` should be `True`

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #36076
- [x] closes #28949
- [x] closes #21020
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_reductions.py [hunk 1]
```diff
@@ -1308,6 +1311,112 @@ def test_frame_any_with_timedelta(self):
...
+    @pytest.mark.parametrize("method", ["any", "all"])
+    def test_any_all_categorical_dtype_nuisance_column(self, method):
+        # GH#36076 DataFrame should match Series behavior
+        ser = Series([0, 1], dtype="category", name="A")
+        df = ser.to_frame()
+
+        # Double-check the Series behavior is to raise
+        with pytest.raises(TypeError, match="does not implement reduction"):
+            getattr(ser, method)()
+        with pytest.raises(TypeError, match="does not implement reduction"):
```
Comment (jreback) [inline pandas/tests/frame/test_reductions.py:1324]: blank line here

File: pandas/tests/frame/test_reductions.py [hunk 2]
```diff
@@ -1308,6 +1311,112 @@ def test_frame_any_with_timedelta(self):
...
+    def test_min_max_categorical_dtype_non_ordered_nuisance_column(self, method):
+        # GH#28948 DataFrame.min should behave like Series.min
+        cat = Categorical(["a", "b", "c", "b"], ordered=False)
+        ser = Series(cat)
+        df = ser.to_frame("A")
+
+        # Double-check the Series behavior
+        with pytest.raises(TypeError, match="is not ordered for operation"):
+            getattr(ser, method)()
+        with pytest.raises(TypeError, match="is not ordered for operation"):
```
Comment (jreback) [inline pandas/tests/frame/test_reductions.py:1378]: blank line

File: doc/source/whatsnew/v1.2.0.rst [hunk 3]
```diff
@@ -284,6 +284,58 @@ of columns could result in a larger :class:`Series` result. See (:issue:`37799`)
     In [6]: df[["B", "C"]].all(bool_only=True)
 
 
+Other :class:`DataFrame` reductions with ``numeric_only=None`` will also avoid
+this pathological behavior (:issue:`37827`):
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:288]: can you make sure to list all the issues you are closing the OP (3?)

**Comment by jreback** [comment]
#28948 is an incorrect ref (top of PR)

**Review by jreback — CHANGES_REQUESTED** [review]
doc comments & there is an oncorrect ref to #28948

**Review by jreback — APPROVED** [review]

File: doc/source/whatsnew/v1.2.0.rst [hunk 4]
```diff
@@ -284,6 +284,63 @@ of columns could result in a larger :class:`Series` result. See (:issue:`37799`)
...
+    Out[3]: Series([], dtype: float64)
+
+    In [4]: df[["A"]].mean()
+    Out[4]:
+    A    1.0
+    dtype: float64
+
+*New behavior*:
+
+.. code-block:: ipython
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:309]: needs to be an ipython block to render

File: doc/source/whatsnew/v1.2.0.rst [hunk 5]
```diff
@@ -284,6 +284,63 @@ of columns could result in a larger :class:`Series` result. See (:issue:`37799`)
...
+
+    In [5]: df.any()
+    Out[5]:
+    A    True
+    dtype: bool
+
+*New behavior*:
+
+.. code-block:: ipython
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:339]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 4 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.2.0.rst:309]: dangit, i thought i had copied from the previous reduction PR, was so proud of myself

Comment (jreback) [reply doc/source/whatsnew/v1.2.0.rst]: haha

**Review by jreback — APPROVED** [review]

**Pull request merged**